### DISCUSSION TITLE
Redirect the user to a differnt page in a server route

### DISCUSSION THREAD

**Discussion by DarraghOFlaherty** [nuxt/nuxt#16966] [Questions]
I am trying to do the following with a nuxt server route, make a call to my database to find the user by id, if the database returns a null indicating the user doesn't exist, immediately redirect the user to the `/not-found` route.

Instead of redirecting the user, the 302 response is returned to the component.

Is this possible? Is it possible to redirect the user before the route is loaded?

My Server Route
```
import { sendRedirect } from "h3";

export default defineEventHandler(async (event) => {
  const { data } = await findById(event.params.id)

  if (!data) {
    return sendRedirect(event, "/not-found");
  }

  return data;
});
```

My Component
```
<template>
  <div class="text-3xl">User: {{ data }}</div>
</template>

<script setup lang="ts">
const route = useRoute();
const { data, pending, error } = await useFetch(
  `/api/users/${route.params.id}`
);
</script>
````

**Comment by mihail727** [comment]
@pi0 can you check this please
